core/trino-main/src/main/java/io/trino/execution/SqlTask.java

```diff
@@ -555,8 +567,14 @@ private SqlTaskExecution tryCreateSqlTaskExecution(Session session, Span stageSp
             // this must happen after taskExecution.start(), otherwise it could become visible to a
             // concurrent update without being fully initialized
             checkState(taskHolderReference.compareAndSet(taskHolder, new TaskHolder(execution)), "unsynchronized concurrent task holder update");
-            return execution;
```

Comment:
So this covers non-final task holder reference set and the notifiy.
Is final task holder reference set also followed up with such notification?

Reply:
[No it is not - but update to ](https://github.com/trinodb/trino/blob/7ebc1c1da7b1753c04ab41af2f01180c3d31b1df/core/trino-main/src/main/java/io/trino/execution/SqlTask.java#L231-L234)